cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java

```diff
@@ -310,15 +312,45 @@ public Boolean apply(StreamEvent input) {
    * @param startTime Timestamp in milliseconds to start reading event from (inclusive)
    * @param endTime Timestamp in milliseconds for the last event to read (exclusive)
    * @param limit Maximum number of events to read
+   * @param results Collection for storing the resulting stream events
+   * @param <T> Type of the collection for storing results
+   * @return The same collection object as passed in the {@code results} parameter
+   * @throws IOException If fails to read from stream
+   * @throws StreamNotFoundException If the given stream does not exists
+   */
+  @Deprecated
+  public <T extends Collection<? super StreamEvent>> T getEvents(String streamId, long startTime,
+                                                                 long endTime, int limit, final T results)
+    throws IOException, StreamNotFoundException, UnauthorizedException {
+    getEvents(streamId, startTime, endTime, limit, new Function<StreamEvent, Boolean>() {
```

Comment:
Shouldn't this one just call the other `getEvents(String, String, String, int, T)` (the one above)?